src/types.ts

```diff
@@ -0,0 +1,122 @@
+// Copyright 2023 the Deno authors. All rights reserved. MIT license.
+
+/**
+ * Common configuration that will be obtained from
+ * environment variables if not explicitly provided.
+ */
+export interface OAuthEnvConfig {
+  /** The client ID provided by the authorization server. */
+  clientId: string;
+
+  /** The client secret provided by the authorization server, if using a confidential client. */
+  clientSecret: string;
+
+  /** The custom domain of the authorization server, if supported by the provider. */
+  domain?: string;
```

Comment:
I've added this, as it's a common case, not juist for auth providers such as Okta & Auth0, but also for services that allow self deployment, eg. GitLab (although the gitlab provider doesn't currently support this, it probably should in the future).